### PR TITLE
gmmk: support VIA

### PR DESCRIPTION
**Pull request by gloryhzw**
## Description
1. Support VIA keymap
2. Optimize rev2 I2C

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/full/rules.mk [hunk 1]
```diff
@@ -39,11 +40,12 @@ OPT_DEFS = -Os
 #
 BACKLIGHT_ENABLE = no
 MAGIC_ENABLE = no
-BOOTMAGIC_ENABLE = yes # Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = yes 	# Virtual DIP switch configuration
```
Comment (dexter93) [inline keyboards/gmmk/full/rules.mk]: tab

**Comment by DaoWen** [comment]
@gloryhzw - Does VIA software work for you after flashing the "via" firmware you merged here? I'm just curious because I tried flashing it to my GMMK Full Rev3, and VIA refused to detect my device.

Regardless, the updates to the keymaps and other changes you made were *very* helpful to me. I don't mind running `qmk` locally, so not having VIA isn't a big deal. After a few iterations and using your patch as an example, I was *finally* able to get my GMMK configured with the hotkey bindings and `RGB_MATRIX_SOLID_REACTIVE` LED scheme that I wanted.

**TL;DR** — Thanks for contributing this patch!

***EDIT:*** Talked to @gloryhzw on discord, and it turns out that I just needed to load the keymap json into VIA, and then it picked it up my device right away. 🤦‍♂️

**Comment by gloryhzw** [comment]
Have a nice time with qmk:)

**Comment by guilleortas** [comment]
I'm getting a segfault error when trying to build any of the variants listed in the [readme](https://github.com/SonixQMK/qmk_firmware/tree/sn32_master/keyboards/gmmk/full#readme) of the GMMK 1 full-size.
I'm using a clean install of QMK MSYS, I can't tell what's going on and I'd really like to build this firmware

**Comment by gloryhzw** [comment]
Try to use the latest qmk msys 1.7.1

**Comment by guilleortas** [comment]
I'm sorry for my blunt message, I'd like to really thank the SonixQMK project for the amazing work you're doing, it's such a powerful resource.
I was already using v1.7.1, and I have tried a clean install on a different computer but the problem is still there. Does it compile for you?

**Comment by guilleortas** [comment]
To update on this matter, removing a few folders you don't need from the keyboards directory seems to do the trick: This is connected to stack size, see [this issue](https://github.com/qmk/qmk_firmware/issues/13416#issuecomment-874362327) for more insight.

**Comment by gloryhzw** [comment]
No issue for 1.7.1 here. You can go sonixqmk discord for help:
https://discord.com/channels/805578807477534750/805578807477534754

guilleortas ***@***.***> 於 2022年8月19日 週五 下午2:00寫道：
